docs/static_assets.rst

```diff
@@ -0,0 +1,206 @@
+  ``cms.envs.json``).
+
+``openedx.core.lib.xblock_pipeline.finder.XBlockPipelineFinder``
+  Custom finder that accesses and extracts assets from pip-installed XBlocks via
+  ``pkg_resources``.
+
+``openedx.core.storage.DevelopmentStorage/ProductionStorage``
+  Custom ``FileStorage`` classes that mostly exist for theme-awareness.
+
+LMS and Studio/CMS Separation
+-----------------------------
+
+LMS and Studio have their own directories for source assets (``lms/static`` and
+``cms/static``), and have symlinks to shared assets in ``common/static``. We
+treat the static asset compilation and collection phase for LMS and Studio as
+separate projects that happen to share a lot of pieces. They output to different
+places (typically ``/edx/var/edxapp/staticfiles`` for LMS and
+``/edx/var/edxapp/staticfiles/studio`` for Studio) and can be collected
+separately. However in practice they're always run together because we deploy
+them from the same commits and to the same servers.
+ 
+Django vs. Webpack Conventions
+******************************
+
+The Django convention for having an app with bundled assets is to namespace them
+locally with the app name so that they get their own directories when they are
+gathered together into a common static directory by collectstatic. For example,
+the edx-enterprise app has a ``static/enterprise`` folder, so its assets are
+compiled to ``/edx/var/edxapp/staticfiles/enterprise`` by edx-platform and will
+not conflict with assets from any other Django app.
+
+Webpack conventions would have us create a single set of configuration files at
+the root of edx-platform, which would specify all bundles in the project.
+
+TODO: The big, "pluggable Webpack components" question.
+
+Proposed Repo Structure
+***********************
+
+All assets that are in common spaces like ``common/static``, ``lms/static``,
+and ``cms/static`` would be moved to be under the Django apps that they are a
+part of and follow the Django naming convention (e.g.
+``openedx/features/course_bookmarks/static/course_bookmarks``). An app's
+``templates/{appname}`` directory will only be for server side templates, and
+any client-side templates will be put in ``static/{appname}/templates``.
+
+Proposed Compiled Structure
+***************************
+
+This is meant to be a sample of the different types of things we'd have, not a
+full list:
+
+::
+
+  # Webpack bundles/post-processed assets
+  /webpack/css
+          /fonts
+          /js
+          /vendor ?
+
+  # Django apps that are in the edx-platform repo
+  /course_bookmarks
+  /course_experience
+
+  # edX authored, installed via separate repo
+  /enterprise
+
+  # Entirely third party apps that we need to maintain compatiblity with.
+  /admin
+  /rest_framework
+
+  # Themes are part of the "theming" app
+  /theming/themes/open-edx
+                 /red-theme
+                 /edx.org
+  
+  # XBlocks still collect their assets into a common space (/xmodule goes away)
```

Comment:
XBlocks have their own conventions, and aren't really managed through Django's staticfiles system. Control and management of that all happens within the custom file finder that lives in edx-platform, so I thought it would make sense to gather all their assets in one place. Or I guess another way to look at it is that the thing managing all the XBlock assets and access to them is the XBlock runtime in edx-platform, and it controls that namespace, and not the individual XBlocks as separate apps. Which models the way XBlocks think about the assets as well (e.g. `self.runtime.resource_url('checkmark.png')`). It also gives us potentially more flexibility in the long term to do cross-XBlock optimizations (e.g. using the same JS libs), or sandboxing/isolation to have that centrally controlled.

Reply:
So I guess it really is "just another app", and that app is called "xblock" or some variant of that.